### PR TITLE
Migrate to `vertx`

### PR DESCRIPTION
**Pull request by tdroxler**
I tried to have a readable diff, I redo what I did in the node, keeping same wording as akka-http-testkit.

Unfortunatelly with `vertx` we have to start the server: `server.listen` in order to test the routes, we can't do as in akka.

So now for all the `web.*ServerSpec` I start once the server on top of the file and all test are using the same instance.

Resolves:  #270

### CODE REVIEW

**Comment by tdroxler** [comment]
we are still using `akka` [ to stop the all system ](https://github.com/alephium/explorer-backend/blob/6189c596cbe991d30de5e4c4b6adf0d8d3d1a4b2/app/src/main/scala/org/alephium/explorer/SyncServices.scala#L101) for sure we can do differently.

We also use `ByteString` but this could be replace by a simple `Array[Byte]`

File: app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  /** Stop AkkaHttp server and terminates ActorSystem */
+  def stopSelfOnce(): Future[Unit] = {
+    for {
+      binding <- httpBindingPromise.future
+      _       <- binding.close().asScala
+    } yield {
+      logger.info(s"http unbound")
+    }
+  }
```
Comment (simerplaha) [inline app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala:85]: Think there is a need close the Vertx instance here with `vertx.close()`? Without it the Thread is held

> [WARN] [10/03/2022 15:35:46.749] [default-akka.actor.internal-dispatcher-2] [CoordinatedShutdown(akka://default)] Coordinated shutdown phase [before-actor-system-terminate] timed out after 5000 milliseconds
2022-10-03 15:35:50,078 [-dispatcher-4] INFO  c.z.h.HikariDataSource                 - db.db - Shutdown completed.
2022-10-03 15:35:50,084 [-dispatcher-5] INFO  o.a.e.ExplorerHttpServer               - http unbound

With `vertx.close()` it properly exists

> 2022-10-03 15:37:41,909 [-dispatcher-6] INFO  c.z.h.HikariDataSource                 - db.db - Shutdown completed.
2022-10-03 15:37:41,936 [dispatcher-11] INFO  o.a.e.ExplorerHttpServer               - http unbound
Process finished with exit code 0

File: app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+/** Stores AkkaHttp related instances created on boot-up */
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+class ExplorerHttpServer(host: String, port: Int, val routes: ArraySeq[Router => Route])(
+    implicit val executionContext: ExecutionContext
+) extends Service
+    with StrictLogging {
+
+  private val vertx  = Vertx.vertx()
+  private val router = Router.router(vertx)
```
Comment (simerplaha) [inline app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala:39]: The issue #356 might be valid here? `vertx` is being created without calling `startSelfOnce`.

Comment (tdroxler) [reply app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala:39]: nice catch, I'll have a look

File: app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala (hunk 1 above)
Comment (tdroxler) [reply app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala:85]: oh nice catch, I didn't thought vertx  needed to be close!

File: app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala (hunk 2 above)
Comment (tdroxler) [reply app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala]: done

File: app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala (hunk 1 above)
Comment (tdroxler) [reply app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala]: done

**Review by simerplaha — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**